src/preferences/preferences.cpp

```diff
@@ -1550,6 +1550,37 @@ const std::vector<game_config::server_info>& prefs::builtin_servers_list()
 	return pref_servers;
 }
 
+void prefs::add_game_preset(config& preset)
+{
+	if(preset.has_child(prefs_list::game_preset) && preset.all_children_count() == 1 && preset.attribute_count() == 0) {
+		int max = 0;
+		for(const auto& c : preferences_.child_range(prefs_list::game_preset)) {
+			max = std::min(max, c["id"].to_int());
```

Comment:
This function is only called from `mp_create_game::save_preset()`,  but there's no `id` key written to the input config there.

Reply:
Right, so first it'll default to -1. Then next time it'll find -1 and so use -2, etc.